### PR TITLE
LPS-180501 [FE] Configuration: Add new Recent (Searches, Pages, Sites, Viewed) types to Site Activities

### PR DESCRIPTION
**Pull request by oliv-yu**
https://liferay.atlassian.net/browse/LPS-180501 - [FE] Configuration: Add new Recent (Searches, Pages, Sites, Viewed) types to Site Activities

New changes are behind a FF:
`feature.flag.LPS-159643=true` (already established for top/trending searches)
`feature.flag.LPS-176691=true` (new FF)

### Contributor Type - Naming
The new contributor types are named like so, prefixed with `asah` to maintain consistency with top/trending search:
Recent Pages: `asahRecentPages`
Recent Searches: `asahRecentSearches` (Trending searches already goes by `asahRecentSearchKeywords` so I wasn’t sure how to make this more distinguishable)
Recent Sites: `asahRecentSites`
Recently Viewed: `asahRecentAssets` (AC API uses term ‘assets’)

For Recently Viewed, the new `contentType` is an array that would accept `blog`, `document`, `form`, and `web-content` (to mirror the AC API `contentType`). If it is set to `[]` the modal will assume all are selected, and the default value is `[]`. The AC team is currently getting back to us [on how their API works](https://liferay.atlassian.net/browse/LRAC-12917?focusedCommentId=2493528), so this might change.

Edit (Note for Backend): The AC team has let us know it's only a singular value for contentType, but the hope is that "Recently Viewed" is able to show multiple types. If this is not possible on the backend, feel free to grab [the latest  commit off this](https://github.com/oliv-yu/liferay-portal/commits/LPS-180501-recent-activities-searchbar-single) to make it a required single String, like how the API goes.

The expectation from backend would be that these suggestions will be available in `/suggestions` API results, given the payload to be something like: 

```
[
    {
        "attributes": {
            "characterThreshold": "0",
            "matchDisplayLanguageId": true
        },
        "contributorName": "asahRecentSearches",
        "displayGroupName": "recent-searches",
        "size": "3"
    },
    {
        "attributes": {
            "characterThreshold": "0"
        },
        "contributorName": "asahRecentPages",
        "displayGroupName": "recent-pages",
        "size": "3"
    },
    {
        "attributes": {
            "characterThreshold": "0"
        },
        "contributorName": "asahRecentSites",
        "displayGroupName": "recent-sites",
        "size": "3"
    },
    {
        "attributes": {
            "characterThreshold": "0",
            "contentType": [
                "blog",
                "form",
                "web-content"
            ]
        },
        "contributorName": "asahRecentAssets",
        "displayGroupName": "recently-viewed",
        "size": "3"
    }
]
```

### Screenshots
Screenshot 2023-10-26 at 10 10 21 AM

Screenshot 2023-10-26 at 10 29 03 AM

Screenshot 2023-10-26 at 10 10 42 AM

Screenshot 2023-10-26 at 10 11 07 AM

### Additional Changes
An extra commit was added to implement the `LearnMessage` component from frontend js components, instead of needing maintain our own component. It shouldn't affect how the component looks and it's somewhat related since new learn links have been added for the new contributor types. However, the learn link beside new Site Activities will not be available since it looks like there is [no URL](https://github.com/liferay-search/liferay-portal/blob/259e93090188cbf1a0ddca3d8d63a4b2d116212e/learn-resources/portal-search-web.json#L22) provided for site activities yet.

Hope that makes sense! Please feel free to make suggestions, thanks for taking a look!

### CODE REVIEW

**Comment by oliv-yu** [comment]
ci:test:sf

**Comment by oliv-yu** [comment]
ci:test:sf

**Comment by oliv-yu** [comment]
ci:test:search

**Comment by oliv-yu** [comment]
ci:test:sf

**Comment by oliv-yu** [comment]
ci:test:search

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/search_bar_configuration_suggestions/contributor_input_set_item/SiteActivities.js [hunk 1]
```diff
@@ -73,14 +146,38 @@ function SiteActivities({index, onBlur, onInputSetItemChange, touched, value}) {
 		});
 	};
 
+	const _handleChangeAttributeValue = (property) => (newValue) => {
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/search_bar_configuration_suggestions/contributor_input_set_item/SiteActivities.js:144]: To keep these 2 functions distinct, let's rename the other function `_handleChangeAttribute` to `_handleChangeAttributeInput`.

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/search_bar_configuration_suggestions/contributor_input_set_item/inputs/ContentTypeInput.js [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+import ClayModal, {useModal} from '@clayui/modal';
+import ClayTable from '@clayui/table';
+import {ClayTooltipProvider} from '@clayui/tooltip';
+import {ManagementToolbar} from 'frontend-js-components-web';
+import {fetch} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
+
+function ContentTypeModal({
+	networkStatus,
+	initialSelectedTypes = [],
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/search_bar_configuration_suggestions/contributor_input_set_item/inputs/ContentTypeInput.js:41]: Thanks for using `initial*` here! 👍

**Review by thektan — APPROVED** [review]
Just a small optional change, but otherwise looks good! Good work!

Also thanks for using the `frontend-js-components-web` LearnMessage component!

**Comment by oliv-yu** [comment]
Commits are a part of https://github.com/liferay-search/liferay-portal/pull/1136

**Pull request closed without merging**